### DISCUSSION TITLE
Tuning summary does not agree with given search space

### DISCUSSION THREAD

**Discussion by rafaelstevenson** [keras-team/keras-tuner#521] [Q&A]
Hello, I am currently using keras tuner, and it seems there is a problem where i used a for loop to define the number of hidden layers as a search space, but the model results summary does not agree with the given 'num_layers' as when 'num_layers'==3 for example, why there are still hyper parameters for 6 hidden layers as i show in the screenshot. 
![image](https://user-images.githubusercontent.com/67007111/115359972-617d8500-a1e9-11eb-861f-e7890d1055ca.png)
![image](https://user-images.githubusercontent.com/67007111/115360027-6f330a80-a1e9-11eb-85bd-e5e9442b78ae.png)

If anybody could help or know the solution would be truly appreciated, thank you

**Comment by haifeng-jin** [comment]
The tuner would tune all the hps that it saw in the previous trials.
Unless the hp is considered not active, which is decided by the `conditional_scope`.
https://keras.io/api/keras_tuner/hyperparameters/#conditional_scope-method
